docs/streams/developer-guide/dsl-topology-naming.html

```diff
@@ -0,0 +1,290 @@
+		          .to("output")
+		</code>
+
+		When running <code>Topology#describe()</code> yields this string:
+
+		<code>
+		Topologies:
+		   Sub-topology: 0
+		    Source: KSTREAM-SOURCE-0000000000 (topics: [input])
+		      --> KSTREAM-FILTER-0000000001
+		    Processor: KSTREAM-FILTER-0000000001 (stores: [])
+		      --> KSTREAM-MAPVALUES-0000000002
+		      <-- KSTREAM-SOURCE-0000000000
+		    Processor: KSTREAM-MAPVALUES-0000000002 (stores: [])
+		      --> KSTREAM-SINK-0000000003
+		      <-- KSTREAM-FILTER-0000000001
+		    Sink: KSTREAM-SINK-0000000003 (topic: output)
+		      <-- KSTREAM-MAPVALUES-0000000002
+		 </code>
+
+		 From this report, we can see what the different operators are, but what is the broader context here?  
+		 For example, consider <code>KSTREAM-FILTER-0000000001</code>, we can see it's a 
+		 filter operation meaning records are dropped that don't match the given predicate.  But what is 
+		 the meaning of the predicate?  Additionally, we can see the topic names of the source and sink nodes, 
+		 but what if the topics aren't named in a meaningful way?  Then we're left to guess the 
+		 business purpose behind these topics.
+
+		 You'll also notice the numbering here; the source node is suffixed with <code>0000000000</code> 
+		 indicating it's the first processor in the topology.  
+		 The filter is suffixed with <code>0000000001</code> indicating it's the second processor in 
+		 the topology.   In the 2.4 release of Kafka Streams, there are now overloaded methods for 
+		 both <code>KStream</code> and <code>KTable</code> accepting 
+		 a new parameter <code>Named</code>.  By using the <code>Named</code> class DSL users can 
+		 now provide meaningful names to the processors in their topology.  
+
+		 Now let's take a look at your topology with naming all the processors:
+		 <code>
+		     builder.&gt;String,String&lt;stream("input", Consumed.as("Customer_transactions_input_topic"))
+		           .filter((k,v) -> !v.equals("invalid_txn"), Named.as("filter_out_invalid_txns"))
+		           .mapValues((v) -> v.substring(0,5), Named.as("Map_values_to_first_6_characters"))
+		           .to("output", Produced.as("Mapped_transactions_output_topic"));
+		 </code>
+
+		 <code>
+		 Topologies:
+		   Sub-topology: 0
+		    Source: Customer_transactions_input_topic (topics: [input])
+		      --> filter_out_invalid_txns
+		    Processor: filter_out_invalid_txns (stores: [])
+		      --> Map_values_to_first_6_characters
+		      <-- Customer_transactions_input_topic
+		    Processor: Map_values_to_first_6_characters (stores: [])
+		      --> Mapped_transactions_output_topic
+		      <-- filter_out_invalid_txns
+		    Sink: Mapped_transactions_output_topic (topic: output)
+		      <-- Map_values_to_first_6_characters
+		 </code>
+
+		Now you can look at the topology description and can easily understand what role each processor 
+		plays in the topology.  But there's another reason for naming your processor nodes when you 
```

Comment:
I personally find it more readable when everything is in its own section, eg the section beginning "But there's another reason..." was moved to the `Changing Names` section whose topic it's actually referring to.  But feel free to leave as-is if you or anyone else disagrees